samza-sql/src/main/java/org/apache/samza/sql/avro/AvroTypeFactoryImpl.java

```diff
@@ -44,37 +45,47 @@ public AvroTypeFactoryImpl() {
   }
 
   public SqlSchema createType(Schema schema) {
+    validateTopLevelAvroType(schema);
+    return convertSchema(schema.getFields(), true);
+  }
+
+  /**
+   * Given a schema field, determine if it is an optional field. There could be cases where a field (like audit header)
```

Comment:
auditHeader may confuse quite a few folks who read this code in open source. Please remove reference to auditHeader and provide a generic example on why this would be needed.

Reply:
Sure, removed the reference.